### PR TITLE
[extension/sigv4authextension] update to use generated status

### PR DESCRIPTION
**Pull request by my-git9**
**Description:** <Describe what has changed.>

This updates the metadata for the extension to include stability information used to generate the status table.

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/19172

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: extension/sigv4authextension/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+type: sigv4authextension
+
+status:
+  class: extension
+  stability: beta
```
Comment (codeboten) [inline extension/sigv4authextension/metadata.yaml:5]:
```suggestion
  stability:
    beta: [extension]
```

File: extension/sigv4authextension/factory.go [hunk 2]
```diff
@@ -30,7 +32,7 @@ const (
 
 // NewFactory creates a factory for the Sigv4 Authenticator extension.
 func NewFactory() extension.Factory {
-	return extension.NewFactory(typeStr, createDefaultConfig, createExtension, component.StabilityLevelBeta)
+	return extension.NewFactory(typeStr, createDefaultConfig, createExtension, metadata.Stability)
```
Comment (codeboten) [inline extension/sigv4authextension/factory.go:35]:
```suggestion
	return extension.NewFactory(typeStr, createDefaultConfig, createExtension, metadata.ExtensionStability)
```

File: extension/sigv4authextension/internal/metadata/generated_status.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+package metadata
+
+import (
+	"go.opentelemetry.io/collector/component"
+)
+
+const (
+	Type      = "sigv4authextension"
+	Stability = component.StabilityLevelBeta
```
Comment (codeboten) [inline extension/sigv4authextension/internal/metadata/generated_status.go:11]:
```suggestion
	Type               = "sigv4authextension"
	ExtensionStability = component.StabilityLevelBeta
```

**Review by codeboten — COMMENTED** [review]
Please rebase on main and generate. I think I've got all the changes in the suggestions, but `make generate` will make sure CI doesnt fail :)

**Comment by my-git9** [comment]
This PR conflicts with https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20326, I will close it.
Thanks for all reviewers!

**Pull request closed without merging**